src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts

```diff
@@ -0,0 +1,99 @@
+import { produce } from "immer";
+import { CombinedFormattedOptionProps, FormattedOption } from "./types";
+
+export type FormattedOptionMap<V1, V2, V3> = Map<
+    string,
+    FormattedOption<V1, V2, V3>
+>;
+
+export namespace NestedDropdownListHelper {
+    export const getInitialDropdown = <V1, V2, V3>(
+        currentItems: FormattedOptionMap<V1, V2, V3>,
+        selectedKeyPath?: string[] | undefined
+    ) => {
+        let keyPath = selectedKeyPath;
+
+        if (!keyPath || !keyPath.length) {
+            keyPath = getInitialSubItem(currentItems);
+        }
+
+        const list = produce(
+            currentItems,
+            (draft: Map<string, FormattedOption<V1, V2, V3>>) => {
+                const targetKey = [];
+                keyPath.forEach((key) => {
+                    targetKey.push(key);
+                    const item = getItemAtKeyPath(draft, targetKey);
+                    item.expanded = true;
+                });
+            }
+        );
+
+        return list;
+    };
+
+    export const getVisibleKeyPaths = <V1, V2, V3>(
+        list: Map<string, CombinedFormattedOptionProps<V1, V2, V3>>,
+        isSearch?: boolean
```

Comment:
is it intended for the keyPaths to be out of order in search? I'm wondering if you can remove this isSearch flag entirely

Reply:
Thanks spotted this one!!!